src/circuit/boolean.rs

```diff
@@ -375,35 +375,44 @@ impl Boolean {
         where E: Engine,
               CS: ConstraintSystem<E>
     {
-        let c = Self::xor(&mut cs, a, b)?;
-
-        match c {
-            Boolean::Constant(false) => {
-                Ok(())
-            },
-            Boolean::Constant(true) => {
-                Err(SynthesisError::Unsatisfiable)
+        match (a, b) {
+            (&Boolean::Constant(a), &Boolean::Constant(b)) => {
+                if a == b {
+                    Ok(())
+                } else {
+                    Err(SynthesisError::Unsatisfiable)
+                }
             },
-            Boolean::Is(ref res) => {
+            (&Boolean::Constant(true), a) | (a, &Boolean::Constant(true)) => {
                 cs.enforce(
-                    || "enforce equals zero",
+                    || "enforce equal to one",
                     |lc| lc,
                     |lc| lc,
-                    |lc| lc + res.get_variable()
+                    |lc| lc + CS::one() - &a.lc(CS::one(), E::Fr::one())
```

Comment:
Oops, sorry for mentioning the rhs of the `-` should be a reference, probably forced you to look at the bellman docs. :)

Reply:
No problem. :) I did notice that `Boolean::lc` was not documented but it's not an urgent issue.